src/components/organisms/notifier/NotificationEventCreate.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { FC, useState } from 'react'
+import { makeStyles, Theme } from '@material-ui/core/styles'
+import Grid from '@material-ui/core/Grid'
+import Select from '@material-ui/core/Select'
+import MenuItem from '@material-ui/core/MenuItem'
+import TextField from '@material-ui/core/TextField'
+import GridRow from 'components/atoms/GridRow'
+import { useForm } from 'react-hook-form'
+import GridItem from 'components/atoms/GridItem'
+import { notifierEventTypeLabels } from 'constants/notifier/strings'
+import { SUPPORTED_EVENTS, SupportedEvent, SupportedEventChannel } from 'config/notifier'
+import NotificationChannelsList from 'components/organisms/notifier/NotificationChannelsList'
+import RoundBtn from 'components/atoms/RoundBtn'
+import { NotifierChannel, NotifierEvent, NotifierEventParam } from 'models/marketItems/NotifierItem'
+import { ContractABIEvent } from 'api/blockscout/interface'
+import getABIEvents from 'api/blockscout'
+import Typography from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import useErrorReporter from 'hooks/useErrorReporter'
+
+const useStyles = makeStyles((theme: Theme) => ({
+  select: {
+    textAlign: 'center',
+    height: 40,
+  },
+  button: {
+    marginTop: theme.spacing(5),
+  },
+  gridRow: {
+    marginTop: theme.spacing(5),
+  },
+}))
+
+type Inputs = {
+  contract: string
+}
+
+type Props = {
+    onAddEvent: (event: NotifierEvent) => void
+    channels?: Array<SupportedEventChannel>
+}
+
+type EventData = {
+  contract: string
+  event: ContractABIEvent
+  eventType: SupportedEvent
+  addedChannels: Array<NotifierChannel>
+}
+
+const NotificationEventCreate: FC<Props> = ({
+  onAddEvent, channels,
+}) => {
+  const { register, handleSubmit, errors } = useForm<Inputs>()
+  const [events, setEvents] = useState<Array<ContractABIEvent>>([])
+  const [eventData, setEventData] = useState<Partial<EventData>>({
+    eventType: 'SMARTCONTRACT' as SupportedEvent,
+  })
+
+  const classes = useStyles()
+  const reportError = useErrorReporter()
+
+  const loadContractEvents = (address: string): void => {
+    getABIEvents(address).then((abiEvents: ContractABIEvent[]) => {
+      setEvents(abiEvents)
+
+      if (abiEvents.length) {
+        setEventData({ ...eventData, event: abiEvents[0] })
+      }
+    }).catch((error) => {
+      reportError({
+        error,
+        id: 'contract-notifier',
+        text: error.text,
+      })
+      setEvents([])
+    })
+  }
+
+  const handleEventChange = ({ target: { value } }) => {
+    setEventData({ ...eventData, event: events.filter((event) => event.name === value)[0] })
```

Comment:
```suggestion
    setEventData({ ...eventData, event: events.find(({name}) => name === value) as EventData})
```
provided we can be sure the event will always be found, which I guess is true considering you used `[0]`.

Reply:
done